src/Compilers/Core/Portable/MetadataReference/MetadataReference.cs

```diff
@@ -121,7 +121,9 @@ public MetadataReference WithProperties(MetadataReferenceProperties properties)
             DocumentationProvider documentation = null,
             string filePath = null)
         {
+#pragma warning disable CA2000 // Dispose objects before losing scope - MetadataImageReference has dispose ownership
```

Comment:
Not a leak since there are no resources involved other than managed memory.

---
In reply to: [171754202](https://github.com/dotnet/roslyn/pull/25179#discussion_r171754202) [](ancestors = 171754202)

Reply:
> Not a leak since there are no resources involved other than managed memory.

@tmat That is still an implementation detail, which is liable to change in future. I feel we should honor IDisposable contract and ensure Dispose is always called on these objects.

Reply:
No, that's not an implementation detail. That's a documented behavior.

Reply:
> Not a leak since there are no resources involved other than managed memory.

This may be the case for some factory methods involved here, but it's not the case for all of them (e.g. the `CreateFromStream` methods).

Reply:
CreateFromStream reads the content of the stream to memory. It does not take ownership of the stream. 

---
In reply to: [171924605](https://github.com/dotnet/roslyn/pull/25179#discussion_r171924605) [](ancestors = 171924605)

Reply:
> CreateFromStream reads the content of the stream to memory. It does not take ownership of the stream.

I don't think the analyzer can correctly detect ownersip transfer through heuristics. https://github.com/dotnet/roslyn-analyzers/issues/1617 tracks improving the analyzer here.